ui/src/app/zones/views/ZoneDetails/ZoneDetailsForm/ZoneDetailsForm.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useEffect } from "react";
+
+import { Row, Col } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import type { RootState } from "app/store/root/types";
+import { actions as zoneActions } from "app/store/zone";
+import zoneSelectors from "app/store/zone/selectors";
+
+type Props = {
+  id: number;
+  closeForm: () => void;
+};
+
+export type CreateZoneValues = {
+  description: string;
+  name: string;
+};
+
+const ZoneForm = ({ id, closeForm }: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const errors = useSelector(zoneSelectors.errors);
+  const saved = useSelector(zoneSelectors.saved);
+  const saving = useSelector(zoneSelectors.saving);
+  const cleanup = useCallback(() => zoneActions.cleanup(), []);
+  const zone = useSelector((state: RootState) =>
+    zoneSelectors.getById(state, Number(id))
+  );
+
+  useEffect(() => {
+    dispatch(zoneActions.fetch());
+  }, [dispatch]);
+
+  let nameValue = null;
+  let descriptionValue = null;
```

Comment:
These should be initialised as `""`, otherwise they're not `string` like what's defined in `CreateZoneValues`, but rather `string | null`.

But what I would actually do is render `null` for this component if `zone` hasn't loaded yet, otherwise `initialValues` gets set twice, once with the empty values and once again for the real values. This effectively means the component has to completely render twice.
``` typescript
if (zone) {
  return (
    <FormikForm<CreateZoneValues>
      initialValues={{
        description: zone.description,
        name: zone.name,
      }}
    />
  );
}
return null;